benchmarks/zap_bench_test.go

```diff
@@ -96,6 +104,19 @@ func BenchmarkZapDisabledLevelsAddingFields(b *testing.B) {
 		}
 	})
 }
+
+func BenchmarkZapDisabledLevelsCheckAddingFields(b *testing.B) {
+	logger := zap.NewJSON(zap.Error, zap.Output(zap.Discard))
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			if m := logger.Check(zap.Info, "Should be discarded."); m != nil {
```

Comment:
optional: you could make the `CheckedMessage` type have a method (`OK() bool`, `Log() bool`, not sure), and that method could also support `nil` as well so you don't have to return a new object.

instead of `m != nil`, we could have `m.OK()`